osu.Game.Tests/Visual/Gameplay/TestSceneSkinnableSound.cs

```diff
@@ -101,5 +107,58 @@ public void TestNonLoopingStopsWithPause()
             AddAssert("sample not playing", () => !sample.Playing);
             AddAssert("sample not playing", () => !sample.Playing);
         }
+
+        [Test]
+        public void TestSkinChangeDoesntPlayOnPause()
+        {
+            DrawableSample sample = null;
+            AddStep("start sample", () =>
+            {
+                skinnableSound.Play();
+                sample = skinnableSound.ChildrenOfType<DrawableSample>().Single();
+            });
+
+            AddAssert("sample playing", () => sample.Playing);
+
+            AddStep("pause gameplay clock", () => gameplayClock.IsPaused.Value = true);
+            AddUntilStep("wait for sample to stop playing", () => !sample.Playing);
+
+            AddStep("trigger skin change", () =>
+            {
+                skinSource.TriggerSourceChanged();
+            });
+
+            AddStep("retrieve new sample", () =>
+            {
+                DrawableSample newSample = skinnableSound.ChildrenOfType<DrawableSample>().Single();
+                Assert.IsTrue(newSample != sample, "Sample still hasn't been updated after a skin change event");
```

Comment:
this is weird. why is there an assert *inside* a step? why couldn't this be an assert step (or, more reliably, a wait step)?

Reply:
Pretty sure there are test scenes using it either here or at framework, but could make it an until step, sure.

Reply:
Was there a reason for doing it this way?

Reply:
done it this way as just quick ensuring logical assertion not exactly part of the test scene but mostly to avoid fake pass because the sample hasn't actually changed, but this seems a bit horrible to read actually, .will change it to assert step when I have time to push the requested changes.

Reply:
right, missed the line below.